File: tennis-game/typescript-fpts-tennis-game/src/TennisGame.ts
import {playTennisGame, welcome} from "./GameInteraction";
import {pipe} from "fp-ts/pipeable";
import {createGame, Game} from "./Data";
import {chain, Task} from "fp-ts/Task";

export const tennisGame: Task<Game> = pipe(
    welcome,
    chain(() => playTennisGame(createGame()))
)

File: tennis-game/swift-bow-tennis-game/swift-bow-tennis-game/TennisGame.swift
//
//  TennisGame.swift
//  swift-bow-tennis-game
//
//  Created by Fabrizio Duroni on 08.09.20.
//  Copyright © 2020 Fabrizio Duroni. All rights reserved.
//

import Foundation
import BowEffects

func tennisGame() -> IO<Never, Game> {
    return welcome().flatMap { playTennisGame(game: createGame()) }^
}
